src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -0,0 +1,279 @@
+            return retval;
+        }
+        retval = iter->second.as_string();
+    }
+    catch (...)
+    {
+    }
+    return retval;
+}
+
+bool fx_muxer_t::resolve_sdk_dotnet_path(const pal::string_t& own_dir, pal::string_t* cli_sdk)
+{
+    pal::string_t cwd;
+    pal::string_t global;
+    if (pal::getcwd(&cwd))
+    {
+        for (pal::string_t parent_dir, cur_dir = cwd; true; cur_dir = parent_dir)
+        {
+            pal::string_t file = cur_dir;
+            append_path(&file, _X("global.json"));
+            if (pal::file_exists(file))
+            {
+                global = file;
+                break;
+            }
+            parent_dir = get_directory(cur_dir);
+            if (parent_dir.empty() || parent_dir.size() == cur_dir.size())
+            {
+                break;
+            }
+        }
+    }
+    pal::string_t retval;
+    if (!global.empty())
+    {
+        pal::string_t cli_version = resolve_cli_version(global);
+        if (!cli_version.empty())
+        {
+            pal::string_t sdk_path = own_dir;
+            append_path(&sdk_path, _X("sdk"));
+            append_path(&sdk_path, cli_version.c_str());
+            if (pal::directory_exists(sdk_path))
+            {
+                retval = sdk_path;
+            }
+        }
+    }
+    if (retval.empty())
+    {
+        pal::string_t sdk_path = own_dir;
+        append_path(&sdk_path, _X("sdk"));
+
+        std::vector<pal::string_t> versions;
+        pal::readdir(sdk_path, &versions);
+        fx_ver_t max_ver(-1, -1, -1);
+        for (const auto& version : versions)
+        {
+            fx_ver_t ver(-1, -1, -1);
+            if (fx_ver_t::parse(version, &ver, true))
+            {
+                max_ver = max(ver, max_ver);
+            }
+        }
+        pal::string_t max_ver_str = max_ver.as_str();
+        append_path(&sdk_path, max_ver_str.c_str());
+        if (pal::directory_exists(sdk_path))
+        {
+            retval = sdk_path;
+        }
+    }
+    cli_sdk->assign(retval);
+    return !retval.empty();
+}
+
+/* static */
+int fx_muxer_t::execute(const int argc, const pal::char_t* argv[])
+{
+    pal::string_t own_path;
+
+    // Get the full name of the application
+    if (!pal::get_own_executable_path(&own_path) || !pal::realpath(&own_path))
+    {
+        trace::error(_X("Failed to locate current executable"));
+        return StatusCode::LibHostCurExeFindFailure;
+    }
+
+    auto own_dir = get_directory(own_path);
+
+    if (argc <= 1)
+    {
+        return StatusCode::InvalidArgFailure;
+    }
+    if (ends_with(argv[1], _X(".dll"), false))
+    {
+        pal::string_t app_path = argv[1];
+
+        if (!pal::realpath(&app_path))
+        {
+            return StatusCode::LibHostExecModeFailure;
+        }
+
+        runtime_config_t config(get_runtime_config_json(app_path));
+        if (config.get_portable())
+        {
+            pal::string_t fx_dir = resolve_fx_dir(own_dir, &config, app_path);
```

Comment:
resolve_fx_dir can return empty fx_dir. We should add a check to bail out for the error case.